python/Ganga/GPIDev/Adapters/IGangaFile.py

```diff
@@ -71,6 +111,14 @@ def put(self):
         """
         raise NotImplementedError
 
+    def copyTo(self, targetPath):
+        """
+        Copy a the file to the local storage using the appropriate file-transfer mechanism
+        Args:
+            targetPath (str): Target path where the file is to copied to
+        """
+        raise NotImplementedError
```

Comment:
I'd personally prefer to have this function check if the file is present in the `localDir` and if not, call an internal method. That way all Files get similar logic without having to implement it each time.